### PR TITLE
Add diff and check modes support for NTP module

### PR DESCRIPTION
**Pull request by mingjunzhang2019**
SUMMARY
- add diff and check modes support for NTP and TACACS modules.
- add some change on get_new_config utility functions.

GitHub Issues
N/A

ISSUE TYPE
- Feature Pull Request

COMPONENT NAME
sonic_ntp, sonic_tacacs_server

OUTPUT
- NTP regression test log: 
[regression_test.log](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/12088333/regression_test.log)
- NTP regression test result: 
[regression-2023-07-18-10-46-56.html.pdf](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/12088336/regression-2023-07-18-10-46-56.html.pdf)
- NTP unit test script:
[ntp.txt](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/12088337/ntp.txt)
- NTP unit test with --diff option log:
[ut_diff.log](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/12088340/ut_diff.log)
- NTP unit test with --diff and --check options log:
[ut_diff_check.log](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/12088345/ut_diff_check.log)
- TACACS regression test log: 
[regression_tacacs.log](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/12163895/regression_tacacs.log)
- TACACS regression test result:
[regression-2023-07-21-09-33-31.html.pdf](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/12163898/regression-2023-07-21-09-33-31.html.pdf)
- TACACS unit test script:
[tacacs.txt](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/12163900/tacacs.txt)
- TACACS unit test with --diff option:
[ut_tacacs_diff.log](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/12163908/ut_tacacs_diff.log)
- TACACS unit test with --diff and --check: 
[ut_tacacs_diff_check.log](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/12163911/ut_tacacs_diff_check.log)

Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility or have provided any relevant "breaking_changes" descriptions in a "fragment" file in the "changelogs/fragments" directory of this repository.
- [x] I have provided a summary for this PR in valid "fragment" file format in the "changelogs/fragments" directory of this repository branch. Reference : [Ansible Change Log Document](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to)

How Has This Been Tested?
Tested with real SONIC system. Did regression and unit test with test script.

### CODE REVIEW

**Review by kerry-meyer — CHANGES_REQUESTED** [review]
The main part of the change set looks good.

Thanks for doing the enhancements for replaced and overridden states in the ntp module while you were doing the other changes.

I have posted a few comments requesting minor changes, mostly for readability.

File: changelogs/fragments/281-playbook-check-diff-modes-for-ntp-module.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+major_changes:
+  - Playbook check and diff modes support for sonic_ntp module (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/281).
```
[the remark this answers is not in the document]
Comment (mingjunzhang2019) [reply changelogs/fragments/281-playbook-check-diff-modes-for-ntp-module.yaml]: Done. Thanks.

File: plugins/module_utils/network/sonic/config/ntp/ntp.py [hunk 2]
```diff
@@ -262,23 +283,23 @@ def _state_overridden(self, want, have):
...
-            commands = want
-            requests = self.get_merge_requests(commands, have)
+            add_commands = want
+            add_requests = self.get_merge_requests(add_commands, have)
 
             if len(requests) > 0:
-                commands = update_states(commands, "overridden")
-            else:
-                commands = []
+                requests.extend(add_requests)
```
[the remark this answers is not in the document]
Comment (mingjunzhang2019) [reply plugins/module_utils/network/sonic/config/ntp/ntp.py:301]: Great finding. Thanks!

File: plugins/module_utils/network/sonic/utils/formatted_diff_utils.py [hunk 3]
```diff
@@ -78,7 +78,7 @@ def __DELETE_SUBCONFIG_ONLY(key_set, command, exist_conf):
     return True, new_conf
 
 
-def __DELETE_OP_DEFAULT(key_set, command, exist_conf):
+def __DELETE_LEAFS_OR_CONFIG_IF_NO_NONE_KEY_LEAF(key_set, command, exist_conf):
```
[the remark this answers is not in the document]
Comment (mingjunzhang2019) [reply plugins/module_utils/network/sonic/utils/formatted_diff_utils.py]: Done Thanks!

File: changelogs/fragments/281-playbook-check-diff-modes-for-ntp-module.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (kerry-meyer) [reply changelogs/fragments/281-playbook-check-diff-modes-for-ntp-module.yaml]: Verified. Thanks.

File: plugins/module_utils/network/sonic/config/ntp/ntp.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (kerry-meyer) [reply plugins/module_utils/network/sonic/config/ntp/ntp.py:301]: Verified.

File: plugins/module_utils/network/sonic/utils/formatted_diff_utils.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (kerry-meyer) [reply plugins/module_utils/network/sonic/utils/formatted_diff_utils.py]: Thanks for adding the headers.

**Review by kerry-meyer — APPROVED** [review]
Thanks for making the additional requested changes.

Approved.

**Pull request merged**